components/automate-ui/src/app/services/projects-filter/projects-filter.reducer.ts

```diff
@@ -70,46 +93,43 @@ export function projectsFilterReducer(
 
 function selectionLabel(options: ProjectsFilterOption[]): string {
   const checkedOptions = options.filter(o => o.checked);
-  const projectOptions = options.filter(o => o.value !== 'unassigned-resources');
+  const projectOptions = options.filter(o => o.value !== UNASSIGNED_PROJECT_ID);
   const checkedProjects = projectOptions.filter(o => o.checked);
-  const hasOnlyProjects = projectOptions.length === options.length;
-  const hasNoneChecked = checkedOptions.length === 0;
-  const hasAllChecked = checkedOptions.length === options.length;
   const hasOneOption = options.length === 1;
   const hasOneChecked = checkedOptions.length === 1;
   const hasOneProjectChecked = checkedProjects.length === 1;
-  const hasSomeProjectsChecked = checkedProjects.length > 1;
-  const hasOnlySomeProjectsChecked = hasSomeProjectsChecked && !hasAllChecked;
+  const hasNoProjectsChecked = checkedProjects.length === 0;
   const hasAllProjectsChecked = checkedProjects.length === projectOptions.length;
-  const hasOnlyAllProjectsChecked = hasOnlyProjects ?
-    hasAllProjectsChecked || hasNoneChecked :
-    hasAllProjectsChecked && !hasAllChecked;
+  const hasSomeProjectsChecked = checkedProjects.length > 1 && !hasAllProjectsChecked;
+  const hasUnassigned = options.filter(o => o.value === UNASSIGNED_PROJECT_ID).length === 1;
+  const hasUnassignedChecked =
+    options.filter(o => o.value === UNASSIGNED_PROJECT_ID && o.checked).length === 1;
 
   if (hasOneOption) {
     return options[0].label;
   }
 
-  if (hasOneChecked) {
-    return checkedOptions[0].label;
+  if (hasAllProjectsChecked && hasUnassignedChecked) {
+    return ALL_RESOURCES_LABEL;
   }
 
-  if (hasOneProjectChecked) {
-    return checkedProjects[0].label;
+  if (hasNoProjectsChecked && hasUnassigned && !hasUnassignedChecked) {
+    return ALL_RESOURCES_LABEL;
   }
 
-  if (hasOnlySomeProjectsChecked) {
-    return 'Multiple projects';
+  if (hasOneChecked || hasOneProjectChecked) {
+    return checkedOptions[0].label;
   }
 
-  if (hasOnlyAllProjectsChecked) {
-    return 'All projects';
+  if (hasSomeProjectsChecked) {
+    return MULTIPLE_PROJECTS_LABEL;
   }
 
-  return 'All resources';
+  return ALL_PROJECTS_LABEL;
```

Comment:
All the changes in the above `selectionLabel` function were to make the unit tests pass.